Mathlib/RingTheory/PowerSeries/WellKnown.lean

```diff
@@ -64,6 +65,87 @@ theorem map_invUnitsSub (f : R →+* S) (u : Rˣ) :
 
 #align power_series.map_inv_units_sub PowerSeries.map_invUnitsSub
 
+section invOneSubPow
+
+variable {S : Type _} [CommRing S] (d : ℕ)
+
+theorem mk_one_pow_eq_mk_choose_add :
+    (@mk S 1) ^ (d + 1) = @mk S fun n => Nat.choose (d + n) d := by
+  induction d with
+  | zero => simp only [Nat.zero_eq, zero_add, pow_one, Nat.choose_zero_right, Nat.cast_one]; rfl
+  | succ d hd =>
+      rw [pow_add, pow_one, mul_comm, show Nat.succ d = d + 1 by rfl, ext_iff]; exact fun n => by
+        rw [hd, coeff_mul]; simp only [coeff_mk, one_mul]; rw [Nat.succ_add, Nat.choose_succ_succ,
+        ← Finset.sum_antidiagonal_choose_add]; simp only [Pi.one_apply, one_mul, Nat.cast_sum]
+
+/--
+The power series `mk fun n => Nat.choose (d + n) d`, whose multiplicative inverse is
+`(1 - X) ^ (d + 1)`.
+-/
+noncomputable def invOneSubPow : S⟦X⟧ˣ where
+  val := mk fun n => Nat.choose (d + n) d
+  inv := (1 - X) ^ (d + 1)
+  val_inv := by
+    rw [← mk_one_pow_eq_mk_choose_add, ← mul_pow,
+      show (mk 1) * (1 - X : S⟦X⟧) = 1 by
+      rw [mul_comm, ext_iff]; exact fun n => by
+        by_cases hn : n = 0
+        · subst hn; simp only [coeff_zero_eq_constantCoeff, map_mul, map_sub, map_one,
+          constantCoeff_X, sub_zero, one_mul, coeff_one, ↓reduceIte]; rfl
+        · simp only [one_mul, map_sub, coeff_mk, coeff_one, hn, ↓reduceIte];
+          rw [show n = n - 1 + 1 by exact (Nat.succ_pred hn).symm, sub_mul];
+          simp only [one_mul, map_sub, coeff_mk, Pi.one_apply, coeff_succ_X_mul, sub_self]]
+    exact one_pow (d + 1)
+  inv_val := by
+    rw [← mk_one_pow_eq_mk_choose_add, ← mul_pow,
+      show (1 - X : S⟦X⟧) * (mk 1) = 1 by
```

Comment:
Aah! If you'd made `PowerSeries.mk_one_mul_one_sub_X` then this would just be `mul_comm` and then an application, saving you 6 lines here.